rpnpy/calculator.py

```diff
@@ -561,6 +563,13 @@ def _tryEvalExec(self, command, modifiers, count):
                                       'whitespace in a command line?')
                     raise CalculatorError(*errors)
                 else:
+                    # If we have new variables, add them to the user variables
+                    # list
+                    variables_after = set(self._variables.keys())
+                    new_variables = variables_after.difference(variables_before)
+                    for var in new_variables:
+                        self._userVariables.append(var)
```

Comment:
Why are the user variables a `list` and not a set? Does that mean that a variable could end up in `self._userVariables` twice?  If you really want it to be a `list` you can replace these two lines with `self._userVariables.extend(new_variables)`.